### PR TITLE
kubernetes-csi: promote testing on 1.20 to mandatory

### PR DESCRIPTION
**Pull request by pohly**
The various components should now pass testing on 1.20.

### CODE REVIEW

File: config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml [hunk 1]
```diff
@@ -188,7 +188,7 @@ presubmits:
           cpu: 2000m
   - name: pull-kubernetes-csi-csi-driver-host-path-1-20-on-kubernetes-1-20
     always_run: true
-    optional: true
+    optional: false
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml:191]: Why is this changed to "false" by default now?

File: config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml [hunk 2]
```diff
@@ -229,13 +229,9 @@ presubmits:
         # docker-in-docker needs privileged mode
         securityContext:
           privileged: true
-      resources:
-        requests:
-          # these are both a bit below peak usage during build
-          # this is mostly for building kubernetes
-          memory: "9000Mi"
```
Comment (xing-yang) [inline config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml:236]: Can you explain why this is removed?

File: config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml (hunk 1 above)
Comment (pohly) [reply config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml:191]: Because we no longer want to tolerate test failures on 1.20.

File: config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml (hunk 2 above)
Comment (pohly) [reply config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml:236]: See https://github.com/kubernetes/test-infra/pull/21468#discussion_r598804630
